### DISCUSSION TITLE
Invalid badge for specific JSON response

### DISCUSSION THREAD

**Discussion by k1o0** [badges/shields#6426] [Q&A] [answered]
We have a custom JSON endpoint that returns badge data to shields.io.  Whenever the tests are pending the badges show up as invalid but I can't figure out what's wrong with the JSON.

Here's the JSON that's returned:
`{"schemaVersion":1,"label":"tests","message":"pending","color":"orange"}`

It results in the right-most badge here (label param is overridden):
![badges](https://user-images.githubusercontent.com/25670666/116514140-4985bf80-a8d3-11eb-955e-1e967cbcefbd.PNG)

It displays correctly when the tests have finished.  Here's an example response that renders correctly:
`{"schemaVersion":1,"label":"tests","message":"348 passed, 71 failed, 6 skipped","color":"red"}`

The only thing I can think that's wrong is the colour but orange is used in the example JSON on the [shields.io site](https://shields.io/endpoint).

Any ideas what could be wrong or how to debug it?  I could try using a different colour for the pending status but I'd prefer to keep it orange.

**Comment by k1o0** [comment]
I've changed the color from 'orange' to 'important' and it still shows up as invalid.  I've looked at the headers and they also look fine (they match the headers when a non-pending response is returned).

**Comment by k1o0** [comment]
Okay it seems to work when the message is anything other than 'pending'.  This seems to be a bug, is 'pending' a magic word?  This is the first time this badge has worked in a long time.  All other badges render without problems.
![badges](https://user-images.githubusercontent.com/25670666/116557901-91bed500-a907-11eb-8efd-5f8e645bed56.png)

**Comment by calebcartwright** [comment]
Thanks for reaching out. To be honest, it's a bit hard to follow the thread here since we're trying to match up screenshots of badges without knowing what the actual badges are. You mentioned that you were overriding the label value defined in the Endpoint response, but I'm also puzzled by the mismatch between the message values of your snippet (all test related) relative to the final "in progress" message.

Would it be possible for you to share the actual badge urls for those badges?

**Reply by k1o0** [reply]
Thanks for the quick response.  This morning I figured out the issue:  In my code a 'pending' message is accompanied by a 201 status code.  Sheilds.io evidently doesn't support such a status.  I've change my code so that even when new tests are triggered it will return a 200 status.

**Comment by k1o0** [comment, marked as answer]
For prosperity: When tests were pending the status code I returned was 201, which shields.io doesn't support.  I fixed this by returning a 200 status instead of 201.

**Reply by calebcartwright** [reply]
Thanks for following up with the result!
